hydra_provider/src/bootstrap.rs

```diff
@@ -0,0 +1,36 @@
+use std::net::SocketAddr;
+
+// use crate::commands::service::CommandService;
+use crate::data_feed_service::serial::{SerialDataFeedService, SerialDataFeedServer};
+use crate::data_feed_service::random::{RandomDataFeedService, RandomDataFeedServer};
+use crate::database_service::DatabaseService;
+
+use tonic::transport::Server;
+use tonic_health::server::health_reporter;
+use tonic_health::pb::health_server::HealthServer;
+use tonic_health::server::HealthService;
+
+// Sets up the gRPC server and loads relevant services
+pub async fn bootstrap(
+	server_port: u32,
+	database_address: String,
+) -> () {
+	let server_address: SocketAddr = format!("[::1]:{}", server_port).parse().unwrap();
+
+	let database_service = DatabaseService::new(&database_address.as_str()).await;
+	let serial_data_feed_service = SerialDataFeedService::new(&database_service);
+	let random_data_feed_service = RandomDataFeedService::new(&database_service);
+
+	let (
+		mut health_reporter, 
+		health_service
+	) = health_reporter();
+
+	health_reporter.set_serving::<HealthServer<HealthService>>().await;
+
+	Server::builder()
+		.add_service(health_service)
+		.add_service(SerialDataFeedServer::new(serial_data_feed_service))
+		.add_service(RandomDataFeedServer::new(random_data_feed_service))
+		.serve(server_address);
```

Comment:
If I'm not mistaken, you need to await asynchronous tasks in tokio otherwise they won't be executed at all

Reply:
Yeah ur right this needs to be awaited but in general so long as the process doesn't exit u can use tokio to spawn async tasks without awaiting them. They just run and end whenever they end without interfering. 

If main finishes before the async tasks process will exit and those async tasks might get aborted afaik. 


https://stackoverflow.com/questions/66895709/how-do-i-spawn-a-long-running-tokio-task-within-another-task-without-blocking-th

Although if there's an error in the async function call we might need to handle it properly